libs/game/game.ts

```diff
@@ -131,32 +134,33 @@ namespace game {
         let background = screen.clone();
         // one last screenshot
         takeScreenshot();
-        
+
```

Comment:
add comment about releasing memory

Reply:
all memory is only released after one additional frame - before that there might be references to the scene from running fibers. Though I guess the main memory usage would be background images, so as long as we clear these (which I think we do in pop scene) we should be fine.

Reply:
Found the issue that was causing crashes, the ``let background = screen.clone()`` just above this snippet seems to be taking up just a tad bit too much memory. I've got a version that works around it in a way that works fine on hardware, but it's hacky / liable to break. See https://github.com/Microsoft/pxt-common-packages/blob/3ebc28fb543fabd06f35e9f1fbf64c9960f2c4fe/libs/game/game.ts#L136-L144

Any ideas on how to avoid doing something like this? If I don't clone it, the screen is cleared after the scene goes away, and the background is just a blank screen

Reply:
I think the problem here is that the fibers won't stop until game.over exits, and the fiber keep references to scenes alive, even if you pop them.

I think we should clear some stuff in scene and event context when it is popped. In particular, try to clear the event handlers, sprite lists, background image. @riknoll ?

I guess I could add a runtime function that would set all object fields to `undefined` (`memzero` essentially), but for now we can just add a method.

Reply:
@jwunderl try Michal suggestion, add a destroy/clean function in scene that set all fields to undefined. It should be called in popScene